commit d3b0c647e2d96527449a7833e4f0dabf7aaa158b
Author: Thomas Leonard <talex5@gmail.com>
Date:   Fri Aug 23 14:11:32 2019 +0100

    Fix GUI linking errors on Debian/testing with OCaml 4.05
    
    Errors were:
    
        warning: Failed to load GTK GUI plugin: error loading shared library: /mnt/_build/default/ocaml/gui_gtk/gui_gtk.cmxs: undefined symbol: caml_mutex_lock
    
    and
    
        warning: Failed to load GTK GUI plugin: error loading shared library: /mnt/_build/default/ocaml/gui_gtk/gui_gtk.cmxs: undefined symbol: camlLwt_preemptive__detach_1503

diff --git a/ocaml/dune b/ocaml/dune
index 83a9f21c..7b6404c9 100644
--- a/ocaml/dune
+++ b/ocaml/dune
@@ -6,7 +6,7 @@
  (ocamlc_flags   -linkall -custom)
  (link_flags  (:include support/extra_objects.sexp))
  (package     0install)
- (libraries   zeroinstall_cli zeroinstall))
+ (libraries   zeroinstall_cli zeroinstall threads))
 
 (executable
  (public_name 0install-runenv.exe)
diff --git a/ocaml/gui_gtk/dune b/ocaml/gui_gtk/dune
index cb8cc9f3..1875e728 100644
--- a/ocaml/gui_gtk/dune
+++ b/ocaml/gui_gtk/dune
@@ -25,7 +25,7 @@ let dune = {|
     (action (run ocamlfind ocamlopt -shared -linkall -linkpkg
 			  -dontlink lwt,lwt.unix
 			  -thread
-			  -package lablgtk2,lwt_glib
+			  -package lablgtk2,lwt_glib,lwt.preemptive
 			  %{deps}
 			  -o %{targets}
     )))
@@ -36,7 +36,7 @@ let dune = {|
     (action (run ocamlfind ocamlc -linkall -linkpkg
 			  -dontlink lwt,lwt.unix
 			  -thread -a
-			  -package lablgtk2,lwt_glib
+			  -package lablgtk2,lwt_glib,lwt.preemptive
 			  %{deps}
 			  -o %{targets}
     )))
